zerver/tests/test_custom_profile_data.py

```diff
@@ -396,9 +398,48 @@ def test_delete(self) -> None:
         self.assertFalse(self.custom_field_exists_in_realm(field.id))
 
     def test_delete_field_value(self) -> None:
+        realm = get_realm("zulip")
+        aaron = self.example_user("aaron")
+        self.login_user(aaron)
+
+        field = CustomProfileField.objects.get(name="Biography", realm=realm)
+        result = self.client_delete(
+            "/json/users/me/profile_data",
+            {
+                "data": orjson.dumps([field.id]).decode(),
+            },
+        )
+        self.assert_json_success(result)
+
+        try_update_realm_custom_profile_field(realm, field, "Biography", editable_by_user=False)
+        self.assertEqual(field.editable_by_user, False)
+        result = self.client_delete(
+            "/json/users/me/profile_data",
+            {
+                "data": orjson.dumps([field.id]).decode(),
+            },
+        )
+        self.assert_json_error(result, "User does not have permission to update the field")
+
         iago = self.example_user("iago")
         self.login_user(iago)
-        realm = get_realm("zulip")
+
+        result = self.client_delete(
+            "/json/users/me/profile_data",
+            {
+                "data": orjson.dumps([field.id]).decode(),
+            },
+        )
+        self.assert_json_success(result)
+
+        field = CustomProfileField.objects.get(name="Phone number", realm=realm)
+        result = self.client_delete(
+            "/json/users/me/profile_data",
+            {
+                "data": orjson.dumps([field.id]).decode(),
+            },
+        )
+        self.assert_json_success(result)
```

Comment:
This block is not required since we already test the case for admin deleting the value for editable field below this.